sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/EquatableRuleBase.cs

```diff
@@ -0,0 +1,46 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2017 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+
+using Microsoft.CodeAnalysis;
+using SonarAnalyzer.Helpers;
+
+namespace SonarAnalyzer.Rules.CSharp
+{
+    public abstract class EquatableRuleBase : SonarDiagnosticAnalyzer
```

Comment:
I don't think it is a good pattern to create a new base class just to share a couple of methods. IMHO, it adds too much complexity to be worth it. Also, if we want to reuse the methods in another class, it'd need to inherit from the class.

Wouldn't it be better if we moved IsValidEqualsMethodSymbol to a helper class? Then IsValidEquatableInterfaceSymbol and EqualsMethodName could be either shared (if suitable) or turned into a private method duplicated in both files. Duplication is bad, but for one-liners it's fine.

WDYT?